### PR TITLE
Relax asyncio_mode type definition; it allows to support pytest 6.1+

### LINKED ISSUE #262: Wrong minimum version of pytest
I am using `pytest` v5.4.3 and I receive the following error when executing test cases:
```
File "D:\.venv\lib\site-packages\_pytest\config\argparsing.py", line 177, in addini
    assert type in (None, "pathlist", "args", "linelist", "bool")
AssertionError: type: string
```

In pytest v5.4.3 the `string` type is not yet added to the assertion, thus the error.

A bit more detail:

The `pytest-asyncio` `plugin` calls the `addini` method like this:

```py
parser.addini(
    "asyncio_mode",
    help="default value for --asyncio-mode",
    type="string",  # <----- Problem
    default="legacy",
)
```

whereas the `addini` method (in `_pytest/config/argparsing.py`) looks like this (in v5.4.3):

```py
    def addini(
        self,
        name: str,
        help: str,
        type: Optional["Literal['pathlist', 'args', 'linelist', 'bool']"] = None,
        default=None,
    ) -> None:
        
        assert type in (None, "pathlist", "args", "linelist", "bool")
        ...
```

So, I think the minimum version of `pytest` should be adjusted

### LINKED ISSUE #263: Expose default value for mode by hook
We have hundreds of microservices that use our plugin to avoid writing an extra boilerplate in each service's repository. We want to set `auto` mode to all services by this plugin. If you expose the default value as hook, our plugin will be able to change it for all services. 

Example from documentation: https://docs.pytest.org/en/latest/how-to/writing_hook_functions.html#using-hooks-in-pytest-addoption

### PR DESCRIPTION
**Pull request by asvetlov**
Fixes #262

### CODE REVIEW

**Review by seifertm — CHANGES_REQUESTED** [review]
I don't think this is enough to support pytest 5.4. I set up a test project using code from this PR and pytest 5.4.3 which produces the following error:
```
$ pytest
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/_pytest/main.py", line 187, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 820, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pluggy/hooks.py", line 308, in call_historic
INTERNALERROR>     res = self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/tmp/venv/lib/python3.9/site-packages/pytest_asyncio/plugin.py", line 111, in pytest_configure
INTERNALERROR>     config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)
INTERNALERROR> AttributeError: 'Config' object has no attribute 'issue_config_time_warning'
```

I'm not opposed to raising the minimum required pytest version. Currently we seem to require at least pytest 6.1.

**Comment by asvetlov** [comment]
Uuups. Thanks for cross-checking.
I'll raise the minimal version and extend the GitHub matrix to check both minimal and latest pytest to make sure that the plugin works well.

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+deps =
+     pytest == 6.2.5  # required for Python 3.10, not bad for others
+commands = make test
+allowlist_externals =
+    make
+
+[testenv:pytest-min]
+extras = testing
+deps =
+     pytest == 6.1.0
```
Comment (asvetlov) [inline tox.ini:19]: Make sure that the plugin works with minimal requirements.

**Review by asvetlov — COMMENTED** [review]
Done, thanks for the review!

File: tests/hypothesis/test_base.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ async def test_can_use_fixture_provided_event_loop(event_loop, n):
     await semaphore.acquire()
 
 
-def test_async_auto_marked(pytester):
-    pytester.makepyfile(
+def test_async_auto_marked(testdir):
+    testdir.makepyfile(
```
Comment (asvetlov) [inline tests/hypothesis/test_base.py:47]: Required for pytest 6.1 as I see. Not a big problem though, the used API is the same.

**Comment by seifertm** [comment]
I like that you added a tox env for the minimum pytests version!

I think the added type annotations may have broken this PR, though. On commit c0ca68692ac30187c525ba054b2daaf4af562eae I get this error:

```
(venv) michael@msnote0 /tmp/test $ pytest
Traceback (most recent call last):
  File "/tmp/test/venv/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 187, in console_main
    code = main()
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 318, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
    config = outcome.get_result()  # type: Config
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1280, in parse
    self._preparse(args, addopts=addopts)
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1172, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/tmp/test/venv/lib/python3.9/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/usr/lib/python3.9/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/assertion/rewrite.py", line 171, in exec_module
    exec(co, module.__dict__)
  File "/tmp/test/venv/lib/python3.9/site-packages/pytest_asyncio/__init__.py", line 3, in <module>
    from .plugin import fixture
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "/tmp/test/venv/lib/python3.9/site-packages/_pytest/assertion/rewrite.py", line 171, in exec_module
    exec(co, module.__dict__)
  File "/tmp/test/venv/lib/python3.9/site-packages/pytest_asyncio/plugin.py", line 456, in <module>
    def event_loop(request: pytest.FixtureRequest) -> Iterator[asyncio.AbstractEventLoop]:
AttributeError: module 'pytest' has no attribute 'FixtureRequest'
```

**Comment by asvetlov** [comment]
@seifertm fair point! I've converted the type annotation for FixtureDef to string, tests on 6.1 are passed now.

**Comment by asvetlov** [comment]
I have no idea why GitHub didn't run checks for yesterday's commits.
Perhaps it was an infrastructure incident.
That's why I missed the `FixtureDef` problem yesterday.
@seifertm thank you for checking! Again!

**Comment by asvetlov** [comment]
I'd like to merge it tomorrow, make 0.17.1 bugfix to get it ready tomorrow at the beginning of working days.

**Review by Tinche — APPROVED** [review]

**Pull request merged**